### ISSUE TITLE
Bug: "Warning: unstable_flushDiscreteUpdates: Cannot flush updates when React is already rendering" after hitting breakpoint if container height 100%

### ISSUE THREAD

**Issue by benrhere** [facebook/react#20917]
If you use CSS to set the height of the react app container to 100% (and the other containing containers as well), then you will get the error "Warning: unstable_flushDiscreteUpdates: Cannot flush updates when React is already rendering" outputted to your developer console after you continue from a breakpoint.

I've got a very simple testcase. This is the create-react-app example, with only the following CSS added:

html {
  height: 100%;
}

#root {
 height: 100%;
}

.App {
  height: 100%;
}

React version: 17.0.1
React DOM version: 17.0.1

Repros in Chrome (89.0.4389.72) and Edge (88.0.705.81)

## Steps To Reproduce

1. download and unzip [hello-world.zip](https://github.com/facebook/react/files/6072375/hello-world.zip) (or create a new app from scratch and add the above CSS to index.css)
2. in the folder created above, run "npm install"
3. run "npm start"
4. Open a web browser to the specified location serving the page.
5. Open developer tools, and set a breakpoint somewhere inside of app.js (you can alternatively add a "debugger" statement inside of your app.js)
![image](https://user-images.githubusercontent.com/1451620/109728370-0aade480-7b84-11eb-8f6c-ad7ca3c4a4bf.png)
6. restart the browser, and after hitting the breakpoint, hit continue
7. Check the console and observe the error output.
![image](https://user-images.githubusercontent.com/1451620/109728430-24e7c280-7b84-11eb-92e7-97a170a575d9.png)

There are no dependencies other than react for this repro.

Link to code example:
[hello-world.zip](https://github.com/facebook/react/files/6072375/hello-world.zip)

## The current behavior
Console outputs error whenever a breakpoint is hit, only when CSS has app container sized at 100%

## The expected behavior
Console should not output error noise regardless of sizing of app container

**Comment by bvaughn** [comment]
This does not repro for me.

**Comment by gaearon** [comment]
This is likely https://github.com/facebook/react/issues/20141.

**Comment by benrhere** [comment]
Hi @bvaughn , I re-confirmed that I can still repro this bug when I pull down the zip and run the steps. I also switched to guest mode in chrome to disable extensions and still get the error. Perhaps it's related to platform? I'm running Windows 10 20H2 OS build 19402.844. Are you able to test on Windows?

Also, if it's helpful, here's the stack trace associated with the error in the console: 

  | console.<computed> | @ | index.js:1
-- | -- | -- | --
  | printWarning | @ | react-dom.development.js:67
  | error | @ | react-dom.development.js:43
  | flushDiscreteUpdates | @ | react-dom.development.js:22343
  | flushDiscreteUpdatesIfNeeded | @ | react-dom.development.js:3768
  | dispatchDiscreteEvent | @ | react-dom.development.js:5886
  | App | @ | App.js:8
  | renderWithHooks | @ | react-dom.development.js:14985
  | mountIndeterminateComponent | @ | react-dom.development.js:17811
  | beginWork | @ | react-dom.development.js:19049
  | beginWork$1 | @ | react-dom.development.js:23940
  | performUnitOfWork | @ | react-dom.development.js:22776
  | workLoopSync | @ | react-dom.development.js:22707
  | renderRootSync | @ | react-dom.development.js:22670
  | performSyncWorkOnRoot | @ | react-dom.development.js:22293
  | scheduleUpdateOnFiber | @ | react-dom.development.js:21881
  | updateContainer | @ | react-dom.development.js:25482
  | (anonymous) | @ | react-dom.development.js:26021
  | unbatchedUpdates | @ | react-dom.development.js:22431
  | legacyRenderSubtreeIntoContainer | @ | react-dom.development.js:26020
  | render | @ | react-dom.development.js:26103
  | (anonymous) | @ | index.js:7
  | ./src/index.js | @ | index.js:18
  | __webpack_require__ | @ | bootstrap:856
  | fn | @ | bootstrap:150
  | 1 | @ | reportWebVitals.js:14
  | __webpack_require__ | @ | bootstrap:856
  | checkDeferredModules | @ | bootstrap:45
  | webpackJsonpCallback | @ | bootstrap:32
  | (anonymous) | @ | main.chunk.js:1

**Comment by benrhere** [comment]
Hi @gaearon , I had the same thought and originally posted a comment on that issue, but the author of that issue suggested I create a new issue. I downloaded his testcase and it doesn't seem like he is performing the steps that I am in this repro. For what it's worth, I have found that in a different app, I can still reproduce this when I _don't_ set the CSS as outlined here, and in those cases I don't know what is triggering this. It's my hope that all of these scenarios have the same root cause, but I don't know.
